### ISSUE TITLE
TTS hangs with chatterbox

### ISSUE THREAD

**Issue by douglasware** [0xShug0/audio.cpp#66]
I am running a series of tests on a long running timer. Checking the output this morning I see that both test machines TTS services become unresponsive independently of each other

1. audiocpp_server serializes all speech on one mutex
ServerState::TimedTaskResult ServerState::run_model(
    LoadedModel & model,
    const engine::runtime::TaskRequest & request) {
    std::lock_guard<std::mutex> lock(model.mutex);
    ensure_model_loaded_locked(model);
    // ...
    auto result = model.offline->run(request);
Every /v1/audio/speech call holds model.mutex for the entire CUDA run.

2. One stuck inference blocks everything behind it
audiocpp_server spawns a detached thread per HTTP client:

std::thread(handle_client, client, std::ref(handler)).detach();
When request 7409215b hung inside CUDA, it kept the mutex. Later requests piled up — I saw 34 worker threads in futex_do_wait (blocked on that mutex).

**Comment by 0xShug0** [comment]
@douglasware Thanks for reporting this. Could you share more details about your setup and test workload? I feel mutex is the amplification mechanism, not the root cause. The session owns mutable state so per-model mutex may be necessary. I'd like identify why the Chatterbox request can hang

**Comment by douglasware** [comment]
I will see if it has anything to do with the model specifically. I was running this on two machines overnight (cuda and rocm) and they both fell over. I sent this issue without a whole lot of investigation. The change/hack in my stack was to try to detect the hung state and restart it. ;)

**Comment by 0xShug0** [comment]
> I will see if it has anything to do with the model specifically. I was running this on two machines overnight (cuda and rocm) and they both fell over. I sent this issue without a whole lot of investigation. The change/hack in my stack was to try to detect the hung state and restart it. ;)

Thanks! No worries. A minimal repro or logs when it happens again would help me track down the root cause.

**Comment by patrickjchen** [comment]
https://github.com/0xShug0/audio.cpp/pull/75 to fix the issue which uses timed_mutext instead of the plain mutex.
@douglasware  However I was not able to reproduce the problem. What model, what data (sound/prompt/...) were you using?
Could you please help testing the code? Thanks!

**Comment by 0xShug0** [comment]
@douglasware #75 merged. Thanks to @patrickjchen for this great robustness improvement. However, the root cause still appears to be Chatterbox hanging, which we can't address without a reproducible failure case. I'll tentatively close the issue for now.
